presto-docs/src/main/sphinx/develop.rst

```diff
@@ -18,5 +18,6 @@ This guide is intended for Presto contributors and plugin developers.
     develop/event-listener
     develop/client-protocol
     develop/worker-protocol
+    develop/sidecar-protocol
```

Comment:
Moving the content out of https://github.com/prestodb/presto/blob/master/presto-docs/src/main/sphinx/develop/worker-protocol.rst works well. 

I'm thinking that because this is a Presto native concern, that the new page should be added to https://github.com/prestodb/presto/blob/master/presto-docs/src/main/sphinx/presto-cpp.rst instead of adding the new page to https://github.com/prestodb/presto/blob/master/presto-docs/src/main/sphinx/develop.rst .